meta-ros-common/recipes-extended/pcl/pcl/0002-Do-not-leak-build-directory-paths-1.12.1.patch

```diff
@@ -0,0 +1,55 @@
+From 4dc28976e88af60e273e670f824cf9b30281e86d Mon Sep 17 00:00:00 2001
+From: Stephen Street <stephen@redrocketcomputing.com>
+Date: Tue, 5 Nov 2024 09:04:23 -0800
+Subject: [PATCH] Do not leak build directory paths in sdk
+
+Signed-off-by: Stephen Street <stephen@redrocketcomputing.com>
+---
+ PCLConfig.cmake.in | 9 +++++----
+ 1 file changed, 5 insertions(+), 4 deletions(-)
+
+diff --git a/PCLConfig.cmake.in b/PCLConfig.cmake.in
+index 1e923a3e9..4a8a3a35e 100644
+--- a/PCLConfig.cmake.in
++++ b/PCLConfig.cmake.in
+@@ -23,6 +23,7 @@ if(POLICY CMP0074)
+   # NEW behavior, we need to push a policy stack.
+   cmake_policy(SET CMP0074 NEW)
+ endif()
++cmake_policy(SET CMP0144 NEW)
+ 
+ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/Modules")
+ 
+@@ -91,9 +92,9 @@ macro(find_boost)
+   if(PCL_ALL_IN_ONE_INSTALLER)
+     set(BOOST_ROOT "${PCL_ROOT}/3rdParty/Boost")
+   elseif(NOT BOOST_INCLUDEDIR)
+-    set(BOOST_INCLUDEDIR "@Boost_INCLUDE_DIR@")
++    set(BOOST_INCLUDEDIR "$ENV{OECORE_TARGET_SYSROOT}/usr/include")
```

Comment:
Hi @sgstreet 
OECORE_TARGET_SYSROOT is defined and used in SDK environment.  It should be null in yocto build time.

Reply:
Yes, I'm aware of this will be empty at yocto build time. My hope, untested at this point, is that non-sdk builds will set BOOST_INCLUDEDIR, EIGEN_ROOT and FLANN_ROOT correctly.  Did you test this PR again some your build? 

Can you suggest an different approach to prevent the leakage of the build directory into the SDK? I can remove this commit and update the sdk colcon build instructions if you want?

Reply:
Turns out that the building images works even when OECORE_TARGET_SYSROOT is empty but present in the PCLConfig.cmake file.  But in the interest of making the least change https://github.com/ros/meta-ros/pull/1241/commits/9f4758807cf336f2b3218829952d1f62c8374efb removes the reference to OECORE_TARGET_SYSROOT.  @robwoolley Thanks for the insights!

Reply:
https://github.com/ros/meta-ros/commit/9f4758807cf336f2b3218829952d1f62c8374efb removes the references to OECORE_TARGET_SYSROOT and works both when building images/sdk and when using the sdk.  Is this sufficient?